src/components/SearchInput/index.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useRef } from 'react';
+import ClassNames from 'classnames';
+
+import styles from '@/components/SearchInput/SearchInput.css';
+
+import IconSearch from '@/assets/images/icons/web/search-simple.svg';
+import IconClose from '@/assets/images/icons/web/close.svg';
+
+type Size = 'large' | 'medium';
+type State = '' | 'valid' | 'invalid';
+
+interface Props {
+  clearInputClick: () => void;
+  description?: string;
+  disabled?: boolean;
+  id: string;
+  label?: string;
+  message?: string;
+  onChange: () => void;
+  onSubmit: () => void;
+  placeholder?: string;
+  state?: State;
+  size: Size;
+  value?: string;
+  variablesClassName?: string;
+}
+
+const SearchInput: React.FC<Props> = props => {
+  const {
+    clearInputClick,
+    description,
+    disabled,
+    id,
+    label,
+    onChange,
+    onSubmit,
+    message,
+    placeholder,
+    size,
+    state,
+    value,
+    variablesClassName,
+    ...searchInputProps
+  } = props;
+  const sizeClass = `search--${size}`;
+  const statusClass = state ? `search--${state}` : '';
+  const sizeInputClass = `search-input--${size}`;
+  const messageValidateClass = state ? `search-input--message-${state}` : '';
+
+  const searchInput = useRef<HTMLInputElement>(null);
+
+  const handleFocus = () => {
+    if (searchInput && searchInput.current) {
+      searchInput.current.focus();
+    }
+  };
+
+  const handleSubmit = event => {
+    if (event.keyCode === 13) onSubmit();
```

Comment:
Could you please put this in a variable? maybe creating a folder names `constants` and a file `keyboard`. This may be useful https://github.com/codelittinc/ay-pia-web/blob/257063b85a20cad0ecb984e7edc707995b14577f/src%2Fconstants%2Fkeyboard.ts#L0-L1